src/GraphRequest.ts

```diff
@@ -49,6 +48,7 @@ export interface URLComponents {
 	path?: string;
 	oDataQueryParams: KeyValuePairObjectStringNumber;
 	otherURLQueryParams: KeyValuePairObjectStringNumber;
+	otherURLQueryOptions: any[];
```

Comment:
The idea of maintaining the otherURLQueryOptions:any[] is to handle any data that is passed by user.
As JS is not typed, I could  pass something like` client.api('/me').query(123)`. The idea is to allow the user input to be forwarded to the service as is and the service should validate the user inputted URL. The other reason is to avoid blocking or rejection while parsing any valid URL pattern which we are not aware of.

Reply:
giving some indication to the consumer of what's accepted is always better than giving no indication at all and letting them try to know whether it actually works or not.
A union type could be used here, and that union type could be extended over time if we want to support "more types" in the future.

Reply:
setting the type of otherURLQueryOptions to string[].